packages/@glimmer/syntax/test/parser-node-test.ts

```diff
@@ -610,6 +610,20 @@ test('disallowed mustaches in the tagName space', function(assert) {
   }, /Cannot use mustaches in an elements tagname: `{{bar` at L1:C6/);
 });
 
+test('disallowed mustaches in closing tag', function(assert) {
+  assert.throws(() => {
+    parse('\nbefore <div></ {{some-modifier}} div>');
+  }, 'Error: Invalid end tag `div` (on line 2). (closing tags cannot contain modifiers).');
+
+  assert.throws(() => {
+    parse('\nbefore <div></div {{some-modifier}}>');
+  }, 'Error: Invalid end tag `div` (on line 2). (closing tags cannot contain modifiers).');
+
+  assert.throws(() => {
+    parse('\nbefore <div><span></span {{some-modifier}}></div>');
+  }, 'Error: Invalid end tag `div` (on line 2). (closing tags cannot contain modifiers).');
```

Comment:
should error point into `span` instead of `div`?

Reply:
@lifeart Thanks for this! You're absolutely right. This comment made me realize my tests weren't actually testing correctly, which lead me down the path the determining how to solve this whole issue much more elegantly.